publisher/internal/analytics/ethereum/analytics.go

```diff
@@ -63,6 +75,9 @@ func New(ctx context.Context, db repository.Repository, opts ...Option) (*Analyt
 
 	ret.eventLogCache = &EventLogCache{cache.New(ret.Options.eventLogCacheExpirationTime, ret.Options.eventLogCachePurgeTime)}
 
+	uniswapLiqPoolsABI = readABI("./internal/analytics/ethereum/Uniswap_Liquidity_Pool_contract.json")
```

Comment:
Consider using `embed` package:

```
 import (
  _ "embed"
)

var (
  //go:embed Uniswap_Liquidity_Pool_contract.json
  uniswapLiquidityPoolABI string
)
```
This will embed the file contents under uniswapLiquidityPoolABI during build time. Very convenient.